web/modules/custom/ai_screening_reports/src/Controller/AiScreeningReportsController.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\ai_screening_reports\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\DependencyInjection\AutowireTrait;
+use Drupal\Core\Url;
+use Drupal\ai_screening_project\Helper\ProjectHelper;
+use Drupal\ai_screening_project_track\Helper\ProjectTrackHelper;
+use Drupal\ai_screening_project_track\ProjectTrackInterface;
+use Drupal\node\NodeInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Returns responses for AI screening reports routes.
+ */
+final class AiScreeningReportsController extends ControllerBase {
+  use AutowireTrait;
+
+  private const array COLOR_CODES = [
+    '#047857',
+    '#aac451',
+    '#a4011e',
+    '#4e6ebe',
+    '#6728c2',
+    '#7622b6',
+    '#243a78',
+    '#8636ee',
+    '#413931',
+    '#27d07d',
+    '#63aade',
+    '#7d5f8e',
+    '#b30b78',
+    '#7f8575',
+    '#ec44d4',
+  ];
+
+  public function __construct(
+    private readonly ProjectTrackHelper $projectTrackHelper,
+    private readonly ProjectHelper $projectHelper,
+  ) {
+  }
+
+  /**
+   * Builds the response for displaying project report.
+   */
+  public function __invoke(NodeInterface $node, Request $request): array {
+    return [
+      '#theme' => 'reports_project',
+      '#data' => [
+        'node' => $node,
+        'request' => $request,
+        'projectHelper' => $this->projectHelper,
+        'projectTrackHelper' => $this->projectTrackHelper,
+      ],
+    ];
+  }
+
+  /**
+   * Builds the response for displaying project track chart.
+   */
+  public function projectTrack(Request $request): array|RedirectResponse {
+    $projectTrackIds = $request->get('project_track_id');
+    $projectTracks = [];
+    $loopCounter = 0;
+    // Ensure proper url parameters: ?project_track_id[]=1&project_track_id[]=3.
+    if (is_array($projectTrackIds)) {
+      $projectData = [
```

Comment:
Consider adding and using a `loadTracks` function in `ProjectTrackHelper` to clean up the code:

``` php
    $projectTracks = $this->projectTrackHelper->loadTracks((array)$request->get('project_track_id'));
    if (!empty($projectTracks)) {
      …
      foreach ($projectTracks as $projectTrack) {
        $projectData['dataset'][$id]['chart'] = [ … ];
      }

      return [
        '#theme' => …,
        …
      ];
    }

    // Bad request
    …
```